shared-libs/transitions/src/transitions/registration.js

```diff
@@ -140,20 +184,30 @@ const getWeeksSinceLMP = doc => {
 };
 
 const setExpectedBirthDate = doc => {
-  const lmp = getWeeksSinceLMP(doc);
-  const start = moment(doc.reported_date).startOf('day');
-  if (lmp === 0) {
-    // means baby was already born, chw just wants a registration.
-    doc.lmp_date = null;
-    doc.expected_date = null;
-  } else {
+  let start;
+
+  if (doc.fields.lmpDate) {
+    start = getLMPDateFromString(doc.fields.lmpDate);
+  }
+  else if (doc.fields.lmpYYYY && doc.fields.lmpMM && doc.fields.lmpDD) {
```

Comment:
The challenge is if the fields are in a different level?  Related to the limitation # 3:

> The form field names should match exactly (they are hard-coded in registration.js):
i. Type A: lmpDate
ii. Type B: lmpYYYY, lmpMM, lmpDD.

What do you think about having a xpath function or an enketo widget to calculate this?

Reply:
Yes, since it is just for SMS forms, will it make more sense if we keep it in something specific to SMS forms such as the [`smsParser`](https://github.com/medic/cht-core/blob/1b887c5bbe2957ebbea6246ebbed824ecb080060/api/src/services/report/smsparser.js#L156)?

Reply:
I think so, I've reply above :)

Reply:
The PR is now ready for review. Most of the work is done in `smsParser` and `validation` now.